joomla/joomla-cms
# [5.1] Convert mod_tags_popular to service provider

### Summary of Changes

Converts the popular tags site module to service provider.

### Testing Instructions

1. Make sure you have site module mod_tags_popular published
2. Add some tags and assign them to several content items
3. You should see module output
4. You should not see module output, when all tags are unpublished/deleted and "Show No results text" module setting is disabled.
5. You should see module output with no results message, when all tags are unpublished/deleted and "Show No results text" module setting is enabled.

![mod_tags_popular](https://github.com/joomla/joomla-cms/assets/59166858/89d78d59-c368-4273-aa7b-72f925b29758)

### Actual result BEFORE applying this Pull Request

Module works as expected

### Expected result AFTER applying this Pull Request

Module works same way as before applying pull request

### Link to documentations
Please select:
